src/samples/coap/lwm2m-bs-server.c

```diff
@@ -0,0 +1,155 @@
+#include <errno.h>
+
+const char *known_clients[] = { "cli1", "cli2", NULL };
+
+static struct sol_blob server_one_addr = {
+    .type = &SOL_BLOB_TYPE_NO_FREE,
+    .parent = NULL,
+    .mem = (void *)"coap://localhost:5683",
+    .size = sizeof("coap://localhost:5683") - 1,
+    .refcnt = 1
+};
+
+static void
+delete_cb(void *data,
+    struct sol_lwm2m_bootstrap_server *server,
+    struct sol_lwm2m_bootstrap_client_info *cinfo, const char *path,
+    enum sol_coap_response_code response_code)
+{
+    const char *name = sol_lwm2m_bootstrap_client_info_get_name(cinfo);
+
+    if (response_code != SOL_COAP_RESPONSE_CODE_DELETED) {
+        fprintf(stderr, "The client %s could not delete the object at %s.\n",
+            name, path);
+        return;
+    }
+
+    printf("The client %s deleted the object at %s.\n", name, path);
+}
+
+static void
+write_cb(void *data,
+    struct sol_lwm2m_bootstrap_server *server,
+    struct sol_lwm2m_bootstrap_client_info *cinfo, const char *path,
+    enum sol_coap_response_code response_code)
+{
+    const char *name = sol_lwm2m_bootstrap_client_info_get_name(cinfo);
+
+    if (response_code != SOL_COAP_RESPONSE_CODE_CHANGED) {
+        fprintf(stderr, "The client %s could not write the object/resource at %s.\n",
+            name, path);
+        return;
+    }
+
+    printf("The client %s wrote the object/resource at %s.\n", name, path);
+}
+
+static void
+bootstrap_cb(void *data,
+    struct sol_lwm2m_bootstrap_server *server,
+    struct sol_lwm2m_bootstrap_client_info *bs_cinfo)
+{
+    const char *name;
+    struct sol_lwm2m_resource sec_server_one[3];
+    int r;
+    uint16_t i;
+
+    name = sol_lwm2m_bootstrap_client_info_get_name(bs_cinfo);
+
+    printf("Client-initiated Bootstrap from %s starting!\n", name);
+
+    sol_lwm2m_bootstrap_server_delete_object_instance(server, bs_cinfo, "/", delete_cb, NULL);
+
+    // Server One's Security Object
+    SOL_LWM2M_RESOURCE_INIT(r, &sec_server_one[0], 0, 1,
+        SOL_LWM2M_RESOURCE_DATA_TYPE_STRING, &server_one_addr);
+    if (r < 0) {
+        fprintf(stderr, "Could init Security Object's [Server URI] resource\n");
+        return;
+    }
+    SOL_LWM2M_RESOURCE_INIT(r, &sec_server_one[1], 1, 1,
+        SOL_LWM2M_RESOURCE_DATA_TYPE_BOOLEAN, false);
+    if (r < 0) {
+        fprintf(stderr, "Could init Security Object's [Bootstrap Server] resource\n");
+        return;
+    }
+    SOL_LWM2M_RESOURCE_INT_INIT(r, &sec_server_one[2], 10, 101);
+    if (r < 0) {
+        fprintf(stderr, "Could init Security Object's [Short Server ID] resource\n");
+        return;
+    }
+
+    sol_lwm2m_bootstrap_server_write(server, bs_cinfo, "/0/1",
+        sec_server_one, sol_util_array_size(sec_server_one), write_cb, NULL);
+
+    for (i = 0; i < sol_util_array_size(sec_server_one); i++)
+        sol_lwm2m_resource_clear(&sec_server_one[i]);
+
+    sol_lwm2m_bootstrap_server_send_finish(server, bs_cinfo);
+}
+
+int
+main(int argc, char *argv[])
```

Comment:
I'll look into it. For now, I've just used something similar to the other samples.